### PR TITLE
Added support for user-tags

### PR DESCRIPTION
**Pull request by BShrivastav**
User tag support added.
Default tag  `creator: ibm-packer-plugin`  is added with every run to intimate creator as packer for the custom image.

### CODE REVIEW

**Review by astha-jain — COMMENTED** [review]
Suggested change

File: builder/ibmcloud/vpc/config.go [hunk 1]
```diff
@@ -126,6 +127,10 @@ func (c *Config) Prepare(raws ...interface{}) ([]string, error) {
 		c.ImageName = fmt.Sprintf("packer-vpc-%d", currentTime.Unix())
 	}
 
+	// check for the endpoint details - create an env variable.
+	// use this URL https://tags.global-search-tagging.cloud.ibm.com
+	// if PKR_GLB_SRCH_TAG_URL
```
[the remark this answers is not in the document]
Comment (ujjwal-ibm) [reply builder/ibmcloud/vpc/config.go]: removed

File: builder/ibmcloud/vpc/step_capture_image.go [hunk 2]
```diff
@@ -97,6 +99,40 @@ func (s *stepCaptureImage) Run(_ context.Context, state multistep.StateBag) mult
...
+
+	optGlbTag := globaltaggingv1.GlobalTaggingV1Options{
+		Authenticator: &core.IamAuthenticator{
+			ApiKey: client.IBMApiKey,
+			URL:    config.IAMEndpoint,
+		},
+	}
+	serviceClientOptions, errOpt := globaltaggingv1.NewGlobalTaggingV1(&optGlbTag)
+	if errOpt != nil {
+		err := fmt.Errorf("[ERROR] Error creating global tagging client options: %s", errOpt)
```
[the remark this answers is not in the document]
Comment (ujjwal-ibm) [reply builder/ibmcloud/vpc/step_capture_image.go]: changed

File: README.md [hunk 3]
```diff
@@ -186,6 +186,7 @@ region | string | IBM Cloud region where VPC is deployed. Required.
 subnet_id | string | The VPC Subnet identifier. Required.
 resource_group_id | string | The resource group identifier to use. If not specified, IBM packer plugin uses `default` resource group.
 security_group_id | string | The security group identifier to use. If not specified, IBM packer plugin creates a new temporary security group to allow SSH and WinRM access.
+tags | list |  List of tags that you want for your image. Tags can help you find the image more easily later. Tags can be made as `key:value` pair or in `label` format. Optional.
```
[the remark this answers is not in the document]
Comment (ujjwal-ibm) [reply README.md]: changed

**Review by ujjwal-ibm — APPROVED** [review]

File: builder/ibmcloud/vpc/config.go [hunk 4]
```diff
@@ -21,6 +21,7 @@ type Config struct {
 	IBMApiKey                 string `mapstructure:"api_key"`
 	Region                    string `mapstructure:"region"`
 	Endpoint                  string `mapstructure:"vpc_endpoint_url"`
+	GTEndpoint                string `mapstructure:"gt_endpoint_url"`
```
[the remark this answers is not in the document]
Comment (astha-jain) [reply builder/ibmcloud/vpc/config.go:24]: Please ghost instead of gt

File: builder/ibmcloud/vpc/step_verify_input.go [hunk 5]
```diff
@@ -145,13 +145,25 @@ func (s *stepVerifyInput) Run(_ context.Context, state multistep.StateBag) multi
...
 	// image check ends
 
+	// usertags validation for blanks.
+	if len(config.ImageTags) > 0 {
+		for i := 0; i < len(config.ImageTags); i++ {
+			if config.ImageTags[i] == "" {
+				err := fmt.Errorf("[ERROR] Invalid user tag \"\", tags can be in `key:value` or `label` format, for example:, tags:\"my_tag\" ")
+				state.Put("error", err)
+				ui.Error(err.Error())
+				return multistep.ActionHalt
```
Comment (astha-jain) [inline builder/ibmcloud/vpc/step_verify_input.go:162]: If tagging failed, can we throw an error and not fail the entire run, I believe `return multistep.ActionHalt` will fail the step. What do you think?

File: developer/README.md [hunk 6]
```diff
@@ -46,6 +46,7 @@ There is a `Makefile` and a `Dockerfile` that automate everything for you.
    **builder variables** |
    vpc_endpoint_url | string | Configure URL for VPC test environments. Optional.
    iam_url | string | Configure URL for IAM test environments. Optional.
+   gt_endpoint_url | string | Configure URL for Global Tagging test environments. Optional.
```
Comment (astha-jain) [inline developer/README.md:49]: lets use ghost_endpoint_url please.

**Review by astha-jain — CHANGES_REQUESTED** [review]

**Review by astha-jain — APPROVED** [review]
Looks good

**Pull request merged**